src/core/lib/gprpp/ref_counted_ptr.h

```diff
@@ -49,6 +49,8 @@ class RefCountedPtr {
   template <typename Y>
   // NOLINTNEXTLINE(google-explicit-constructor)
   RefCountedPtr(RefCountedPtr<Y>&& other) noexcept {
+    static_assert(std::has_virtual_destructor<T>::value,
```

Comment:
This could potentially break things, because [`RefCounted<>` supports objects with non-virtual dtors](https://github.com/grpc/grpc/blob/024a0f738898c5440dc4fa0102f72e6712cbb748/src/core/lib/gprpp/ref_counted.h#L259).

What's the benefit that we would get from this?  It's not clear to me that it actually buys us anything, and given the problem I just mentioned, I think it would be better to just remove it.